sdv/tabular/base.py

```diff
@@ -712,16 +739,21 @@ def sample_remaining_columns(self, known_columns, max_tries=100, batch_size_per_
 
         self._randomize_samples(randomize_samples)
 
+        known_columns = known_columns.copy()
+        sampled = pd.DataFrame()
         try:
-            with tqdm(total=len(known_columns)) as progress_bar:
-                sampled = self._sample_with_conditions(
+            def _sample_function(progress_bar=None):
+                return self._sample_with_conditions(
                     known_columns, max_tries, batch_size_per_try, progress_bar, output_file_path)
 
-        except (Exception, KeyboardInterrupt) as e:
-            print('Error: Sampling terminated. Partial results are stored in a temporary file: '
-                  f'{output_file_path}. This file will be overridden the next time you sample. '
-                  'Please rename the file if you wish to save these results.')
-            raise e
+            if len(known_columns) == 1 and max_tries == 1:
+                sampled = _sample_function()
+            else:
+                sampled = progress_bar_wrapper(
+                    _sample_function, len(known_columns), 'Sampling remaining columns')
```

Comment:
Can we add some unit tests to make sure this logic of only making the progress bar under certain conditions is working?

Reply:
Added asserts to the existing unit tests!